### PR TITLE
ci(sdk): Automate merging SDK doc PR

### LINKED ISSUE #43746: Automatically merge the doc PR

### PR DESCRIPTION
**Pull request by WiNloSt**
closes #43746

[The workflow](https://github.com/metabase/metabase/actions/workflows/release-embedding-sdk.yml)

This change will automatically approve the SDK doc PR and set it to auto-merge

### CODE REVIEW

File: .github/workflows/release-embedding-sdk.yml [hunk 1]
```diff
@@ -92,13 +92,6 @@ jobs:
         run: |
           ./bin/embedding-sdk/release_utils.bash update_package_json_template ${{ inputs.sdk_version }}
 
-      # TODO: Remove this step after the build utils script is merged to master
```
Comment (WiNloSt) [inline .github/workflows/release-embedding-sdk.yml:95]: This was needed when we hadn't merged the first workflow PR to `master`. Backend `bin/embedding-sdk/release_utils.bash` didn't exist on master. That was why this step was needed, but not anymore.

File: .github/workflows/release-embedding-sdk.yml [hunk 2]
```diff
@@ -171,13 +164,6 @@ jobs:
           git config --global user.email "github-automation@metabase.com"
           git config --global user.name "Metabase Automation"
 
-      # TODO: Remove this step after the build utils script is merged to master
```
Comment (WiNloSt) [inline .github/workflows/release-embedding-sdk.yml:174]: Same as the previous comment. The previous step updates the util and this step downloads the util

File: .github/workflows/release-embedding-sdk.yml [hunk 3]
```diff
@@ -214,12 +200,24 @@ jobs:
...
           private-key: ${{ secrets.METABASE_BOT_APP_PRIVATE_KEY }}
 
-      - name: Edit PR adding useful information
+      - name: Edit PR adding useful information (using Metabase bot app)
         run: |
           gh pr edit --add-reviewer @metabase/embedding --add-label no-backport
         env:
           GH_TOKEN: ${{ steps.app-token.outputs.token }}
 
+      - name: Auto approve PR (using GitHub Actions account)
```
Comment (WiNloSt) [inline .github/workflows/release-embedding-sdk.yml:209]: I tested these 2 step separately, so we don't accidentally merge the PR to master.

Here's the [test for this step.](https://github.com/metabase/metabase/pull/43814#pullrequestreview-2104227524)

You can see that the auto approval works.

File: .github/workflows/release-embedding-sdk.yml [hunk 4]
```diff
@@ -214,12 +200,24 @@ jobs:
...
+        run: |
+          gh pr review --approve
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Enable Pull request auto-merge (using GitHub Metabase Automation account)
```
Comment (WiNloSt) [inline .github/workflows/release-embedding-sdk.yml:215]: Here's the [test for auto-merge](https://github.com/metabase/metabase/pull/43818#event-13078972668)

**Review by heypoom — APPROVED** [review]
Checked the test results, looks good!

**Pull request merged**